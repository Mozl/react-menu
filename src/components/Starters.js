import React from 'react';
import styled from 'styled-components';
import Dish from './Dish';

const StartersWrapper = styled.div`
  width: 50%;
  display: flex;
  border: 0.1rem solid black;
  flex-direction: column;
  text-color: black;
  border-radius: 1.25rem;
  padding: 1rem;
  margin-top: 1rem;
`;

StartersWrapper.displayName = 'StartersWrapper';

class Starters extends React.Component {
  render() {
    const starters = this.props.dishes.starters;
    return (
      <StartersWrapper>
        {starters.map(dish => (
          <Dish key={dish.id} dishName={dish.name} />
        ))}
      </StartersWrapper>
    );
  }
}
export default Starters;
