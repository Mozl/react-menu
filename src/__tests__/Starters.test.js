import React from 'react';
import ReactDOM from 'react-dom';
import Starters from '../components/Starters';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import menuData from '../menu-data';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Starters dishes={menuData} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer.create(<Starters dishes={menuData} />);
  expect(tree).toMatchSnapshot();
});

it('renders starter dishes', () => {
  const wrapper = mount(<Starters dishes={menuData} />);
  expect(wrapper.find('Dish').length).toEqual(4);
});
