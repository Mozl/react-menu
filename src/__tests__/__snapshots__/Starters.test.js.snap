// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bwzfXH gokkPN"
>
  <div
    className="sc-bdVaJa fVezFz"
  >
    Soup
  </div>
  <div
    className="sc-bdVaJa fVezFz"
  >
    Pâté
  </div>
  <div
    className="sc-bdVaJa fVezFz"
  >
    Bruschetta
  </div>
  <div
    className="sc-bdVaJa fVezFz"
  >
    Prawn cocktail
  </div>
</div>
`;
